gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <string_view>
+
+#include "X86RegisterInfo.h"
+#include "X86Subtarget.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "gematria/datasets/bhive_importer.h"
+#include "gematria/datasets/find_accessed_addrs.h"
+#include "gematria/llvm/canonicalizer.h"
+#include "gematria/llvm/llvm_architecture_support.h"
+#include "gematria/utils/string.h"
+
+constexpr uint64_t kInitialRegVal = 0x10000;
+constexpr uint64_t kInitialMemVal = 0x7FFFFFFF;
+constexpr std::string_view kRegDefPrefix = "# LLVM-EXEGESIS-DEFREG ";
+constexpr std::string_view kMemDefPrefix = "# LLVM-EXEGESIS-MEM-DEF ";
+constexpr std::string_view kMemMapPrefix = "# LLVM-EXEGESIS-MEM-MAP ";
+constexpr std::string_view kMemNamePrefix = "MEM";
+
+namespace {
+unsigned int file_counter = 0;
+}
```

Comment:
I'd keep this as a local variable in `main()`. It's only really used from there, so there's no need to make it global.

Reply:
I've merged the PR as is to expedite it, and I'll fix this myself afterwards.